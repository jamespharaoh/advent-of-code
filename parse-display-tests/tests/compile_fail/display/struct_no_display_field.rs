use parse_display::Display;

#[derive(Display)]
struct TestStruct {
    x: NoDisplay,
}
struct NoDisplay;

fn main() {}
