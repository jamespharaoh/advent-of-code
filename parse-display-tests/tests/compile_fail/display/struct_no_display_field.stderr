error[E0277]: `NoDisplay` doesn't implement `std::fmt::Display`
 --> $DIR/struct_no_display_field.rs:3:10
  |
3 | #[derive(Display)]
  |          ^^^^^^^ `NoDisplay` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `NoDisplay`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required by `std::fmt::Display::fmt`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
